test/unit/potreelayerparsing.js

```diff
@@ -90,18 +90,21 @@ describe('Potree Provider', function () {
                 octreeDir: 'eglise_saint_blaise_arles',
             },
         });
-        const p4 = new PotreeLayer('pointsCloud4', { source, crs: view.referenceCrs });
+        const layer4 = new PotreeLayer('pointsCloud4', { source, crs: view.referenceCrs });
 
-        ps.push(p4);
-        p4.whenReady.then((l) => {
-            assert.ok(!l.material.defines.NORMAL);
-            assert.ok(!l.material.defines.NORMAL_SPHEREMAPPED);
-            assert.ok(l.material.defines.NORMAL_OCT16);
-        });
+        layers.push(layer4);
+        const p4 = layer4.whenReady
+            .then((l) => {
+                assert.ok(!l.material.defines.NORMAL);
+                assert.ok(!l.material.defines.NORMAL_SPHEREMAPPED);
+                assert.ok(l.material.defines.NORMAL_OCT16);
+            });
 
-        ps.forEach(p => View.prototype.addLayer.call(view, p));
+        layers.forEach(p => View.prototype.addLayer.call(view, p));
 
-        Promise.all(ps.map(p => p.whenReady)).then(() => done());
+        Promise.all([p1, p2, p3, p4])
+            .then(() => done())
+            .catch(done);
```

Comment:
you could write

`Promise.all([p1, p2, p3, p4]).then(done, done)`

Reply:
Done

Reply:
`Promise.all([p1, p2, p3, p4]).then(done).catch(done)` ?

Reply:
neither as then(done) is short for then ((res) => done(res)) or done should be call with no parameters and Promise.all return an array.

Reply:
ok